third_party/terraform/resources/resource_dataflow_job.go

```diff
@@ -616,3 +651,29 @@ func waitForDataflowJobToBeUpdated(d *schema.ResourceData, config *Config, repla
 		}
 	})
 }
+
+// toSubnetworkSelfLink converts the given string to a subnetwork self-link of the format:
+//   https://www.googleapis.com/compute/v1/projects/{project}/regions/{region}/subnetworks/{subnetwork_name}
+// The following input formats are supported:
+// - https://www.googleapis.com/compute/ANY_VERSION/projects/{project}/regions/{region}/subnetworks/{subnetwork_name}
+// - projects/{project}/regions/{region}/subnetworks/{subnetwork_name}
+// - regions/{region}/subnetworks/{subnetwork_name}
+// - {subnetwork_name}
+// - "" (empty string). toSubnetworkSelfLink will return an empty string in this case.
+func toSubnetworkSelfLink(subnetwork string, d *schema.ResourceData, config *Config) (string, error) {
+	fv, err := ParseSubnetworkFieldValue(subnetwork, d, config)
+	if err != nil {
+		return "", err
+	}
+	if fv.RelativeLink() == "" {
+		return "", nil
+	}
+	// Dataflow only respects the legacy compute base path with domain
+	// www.googleapis.com, not compute.googleapis.com
+	legacyComputeBasePath := "https://www.googleapis.com/compute/beta/"
```

Comment:
I don't think that we want to hard code the beta version in here. Why do we need to include the full path rather than `projects/{project}/regions/{region}/subnetworks/{subnetwork_name}`?